### PR TITLE
Wallet send from outpoints

### PR DESCRIPTION
**Pull request by benthecarman**
Useful for coin selection

### CODE REVIEW

File: app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala [hunk 1]
```diff
@@ -505,6 +561,12 @@ object CliCommand {
       amount: Bitcoins,
       satoshisPerVirtualByte: Option[SatoshisPerVirtualByte])
       extends CliCommand
+  case class SendFromOutPoints(
+      outPoints: Vector[TransactionOutPoint],
+      destination: BitcoinAddress,
+      amount: Bitcoins,
+      satoshisPerVirtualByte: Option[SatoshisPerVirtualByte])
```
Comment (Christewart) [inline app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala:568]: Do we want this to be available on segwit only? Isn't `sats/vbyte` only available on segwit outpoints?

File: wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala [hunk 2]
```diff
@@ -205,6 +207,59 @@ abstract class Wallet
     } yield updatedInfos
   }
 
+  private def finishSend(txBuilder: BitcoinTxBuilder): Future[Transaction] = {
```
Comment (Christewart) [inline wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala:213]: scaladoc of what this method is doing

File: wallet-test/src/test/scala/org/bitcoins/wallet/WalletSendingTest.scala [hunk 3]
```diff
@@ -140,4 +141,27 @@ class WalletSendingTest extends BitcoinSWalletTest {
       sendToAddressesF
     }
   }
+
+  it should "" in { fundedWallet =>
```
Comment (Christewart) [inline wallet-test/src/test/scala/org/bitcoins/wallet/WalletSendingTest.scala]: name for the test case

File: app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala (hunk 1 above)
Comment (benthecarman) [reply app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala:568]: No, on non-segwit txs it will just be the same as sats/byte

Comment (Christewart) [reply app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala:568]: Ok, what about mixed outpoints? I.e. non segwit and segwit utxos? We probably need to come up with a coherent model for this case -- it's hard to think about!

Comment (benthecarman) [reply app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala:568]: This is just handled by `FeeUnit.calc` should be handled correctly already

Comment (Christewart) [reply app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala:568]: Ok, so when doing our API, does fee unit of `sat/vbyte` or `sat/byte` matter? Can you pay a fee too large or too small of a fee depending on that unit, and how we interpret it?

Obviously if we were to interpret the number as `SatsPerKiloByte` will effect your fee rate, but the difference between `sats/vbyte` & `sats/byte` won't?

Comment (benthecarman) [reply app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala:568]: It's handled correctly, for `sats/vbyte` it does `tx.vsize * fee rate`, for `sats/byte` it does `tx.baseSize * fee rate`

**Review by Christewart — APPROVED** [review]

**Pull request merged**